pkg/config/v3/config.go

```diff
@@ -271,6 +271,18 @@ func (c cfg) HasGroup(group string) bool {
 	return false
 }
 
+// ListResourceswithGK implements config.Config
+func (c cfg) ListResourceswithGK(gvk resource.GVK) []resource.Resource {
+	resources := make([]resource.Resource, 0)
+
+	for _, r := range c.Resources {
+		if r.Group == gvk.Group && r.Kind == gvk.Kind {
+			resources = append(resources, r)
+		}
+	}
```

Comment:
Creating a copy is important to prevent the caller to modify the internal list of resources without knowing it. We also need to restore the plural form in case it was omitted.
```suggestion
	for _, res := range c.Resources {
		if res.Group == gvk.Group && res.Kind == gvk.Kind {
			r := res.Copy()

			// Plural is only stored if irregular, so if it is empty recover the regular form
			if r.Plural == "" {
				r.Plural = resource.RegularPlural(r.Kind)
			}

			resources = append(resources, r)
		}
	}
```